src/index.js

```diff
@@ -137,7 +137,9 @@ export default function promiseMiddleware(config = {}) {
        *   }
        * }
        */
-      return promise.then(handleFulfill, handleReject);
```

Comment:
I prefer that, but it's actually a slightly different meaning (I purposely did this to preserve the behavior of your library)

With the old way, the returned promise does not resolve or reject until the `dispatch` call finishes. In the new way, the promise is resolved before the actions are dispatched (and state updated), which is undesirable.